clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -129,16 +129,12 @@ public boolean isComplete() {
 
     @Override
     public byte[] unwrap(byte[] incoming, int offset, int len) {
-        if (!isComplete())
-            throw new IllegalStateException("Authentication exchange has not completed");
-        return Arrays.copyOfRange(incoming, offset, offset + len);
+        throw new IllegalStateException("Mechanism does not support integrity nor confidentality");
     }
 
     @Override
     public byte[] wrap(byte[] outgoing, int offset, int len) {
-        if (!isComplete())
-            throw new IllegalStateException("Authentication exchange has not completed");
-        return Arrays.copyOfRange(outgoing, offset, offset + len);
+        throw new IllegalStateException("Mechanism does not support integrity nor confidentality");
```

Comment:
I think you missed the change in `OAuthBearerSaslServer`

Reply:
Thank you, fixed.